indico/modules/events/editing/client/js/management/ManageService.jsx

```diff
@@ -0,0 +1,228 @@
+                  to disconnect and reconnect will most likely not fix it.
+                </Param>
+              </Translate>
+            ) : (
+              <Translate>
+                Do you really want to disconnect from the editing workflow service? If your editing
+                process has already started (e.g. people submitted content to be edited),
+                disconnecting is most likely not a good idea and should only be done if you do not
+                plan to enable the service again for this event.
+              </Translate>
+            )}
+          </Modal.Content>
+        }
+      />
+      <Modal size="small" open={connectOpen}>
+        <FinalForm onSubmit={connect} subscription={{}}>
+          {fprops => (
+            <>
+              <Modal.Header content={Translate.string('Connect to editing workflow service')} />
+              <Modal.Content>
+                <Form id="connect-service-form" onSubmit={fprops.handleSubmit}>
+                  <FinalInput
+                    name="url"
+                    label={Translate.string('Service URL')}
+                    description={
+                      <Translate>
+                        Provide the URL for the service implementing your custom editing workflow.
+                      </Translate>
+                    }
+                    autoFocus
+                    required
+                    validate={async value => {
+                      setServiceURLInfo(null);
+
+                      const validationError = v.required(value) || v.url(value);
+                      if (validationError) {
+                        return validationError;
+                      }
+
+                      let resp;
+                      try {
+                        // this does send a request on each keypress, but in this particular case
+                        // it's VERY likely that people paste a url, so it shouldn't be an issue
+                        resp = await indicoAxios.get(checkServiceURL({confId: eventId}), {
+                          params: {url: value},
+                        });
+                      } catch (err) {
+                        // the url validation is not exactly the same on the client and server
+                        // side, so we have some edge cases (e.g. `http://test:`) which pass
+                        // the client-side check but not the server-side one.
+                        // in this case we do not want an error dialog to show up but rather
+                        // fail validation nicely.
+                        const submitError = handleSubmitError(err);
+                        return submitError.url || submitError[FORM_ERROR];
+                      }
+
+                      if (resp.data.error) {
+                        return resp.data.error;
+                      }
+
+                      setServiceURLInfo(resp.data.info);
```

Comment:
If I close the modal, the state of this variable will be retained. It should probably be cleared `onClose`.

Reply:
I don't think `onClose` triggers when using the controlled version of the Modal. Anyway, fixed it and also added the missing onClose when prevented you from closing the modal by clicking e.g. on the dimmer outside.